app/src/main/kotlin/org/cru/godtools/ui/dashboard/lessons/LessonFilters.kt

```diff
@@ -0,0 +1,144 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import java.util.Locale
+import org.ccci.gto.android.common.androidx.compose.material3.ui.list.ListItemStartPadding
+import org.ccci.gto.android.common.androidx.compose.material3.ui.menu.LazyDropdownMenu
+import org.cru.godtools.R
+import org.cru.godtools.base.LocalAppLanguage
+import org.cru.godtools.base.ui.theme.GodToolsTheme
+import org.cru.godtools.model.Language
+import org.cru.godtools.ui.languages.LanguageName
+
+private val DROPDOWN_LESSON_MAX_HEIGHT = 700.dp
+private val DROPDOWN_LESSON_MAX_WIDTH = 350.dp
+
+@Composable
+fun LessonFilters(modifier: Modifier = Modifier) {
+    Row(modifier = modifier) {
+        Text(
+            stringResource(R.string.dashboard_lessons_section_filter_label),
+            modifier = Modifier
+                .alignByBaseline()
+                .padding(end = 36.dp)
+        )
+        LessonLanguageFilter(
+            modifier = Modifier
+                .weight(1f)
+                .alignByBaseline()
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+@Composable
+internal fun LessonLanguageFilter(modifier: Modifier = Modifier, viewModel: LessonsViewModel = viewModel()) {
+    val context = LocalContext.current
+    val selectedLanguage by viewModel.selectedLanguage.collectAsState(initial = Language(Locale.getDefault()))
+
+    val lessonLanguages by viewModel.filteredLanguages.collectAsState(emptyList())
+    var expanded by remember { mutableStateOf(false) }
+
+    ElevatedButton(
+        onClick = { expanded = true },
+        modifier = modifier.semantics { role = Role.DropdownList }
+    ) {
+        Text(
+            selectedLanguage.getDisplayName(context, LocalAppLanguage.current),
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            modifier = Modifier.weight(1f),
+        )
+        ExposedDropdownMenuDefaults.TrailingIcon(expanded = expanded)
+        LazyDropdownMenu(
+            expanded = expanded,
+            onDismissRequest = { expanded = false },
+            modifier = Modifier.sizeIn(maxHeight = DROPDOWN_LESSON_MAX_HEIGHT, maxWidth = DROPDOWN_LESSON_MAX_WIDTH)
+        ) {
+            stickyHeader {
+                Surface(color = MaterialTheme.colorScheme.surface) {
+                    val query by viewModel.query.collectAsState("")
```

Comment:
https://github.com/user-attachments/assets/b357276e-180f-4d57-bfc7-fe7346dba513

the query is not being reset when a selection is made because the state is being tracked outside of the composition scope (in the ViewModel).

You will need to clear the query either when you open the menu, or after you make a selection.